src/legacy/ui/ui_render/ui_render_mixin.js

```diff
@@ -191,101 +174,25 @@ export function uiRenderMixin(kbnServer, server, config) {
     },
   });
 
-  async function getUiSettings({ request, includeUserProvidedConfig }) {
-    const uiSettings = request.getUiSettingsService();
-    return props({
-      defaults: uiSettings.getRegistered(),
-      user: includeUserProvidedConfig && uiSettings.getUserProvided(),
-    });
-  }
-
-  function getLegacyKibanaPayload({ app, basePath, uiSettings }) {
-    return {
-      app,
-      bundleId: `app:${app.getId()}`,
-      nav: server.getUiNavLinks(),
-      version: kbnServer.version,
-      branch: config.get('pkg.branch'),
-      buildNum: config.get('pkg.buildNum'),
-      buildSha: config.get('pkg.buildSha'),
-      serverName: config.get('server.name'),
-      devMode: config.get('env.dev'),
-      basePath,
-      uiSettings,
-    };
-  }
-
   async function renderApp({
-    app,
+    app = { getId: () => 'core' },
     h,
     includeUserProvidedConfig = true,
     injectedVarsOverrides = {},
   }) {
-    const request = h.request;
-    const basePath = request.getBasePath();
-    const uiSettings = await getUiSettings({ request, includeUserProvidedConfig });
-    app = app || { getId: () => 'core' };
-
-    const legacyMetadata = getLegacyKibanaPayload({
-      app,
-      basePath,
-      uiSettings,
-    });
-
-    // Get the list of new platform plugins.
-    // Convert the Map into an array of objects so it is JSON serializable and order is preserved.
-    const uiPluginConfigs = kbnServer.newPlatform.__internals.uiPlugins.browserConfigs;
-    const uiPlugins = await Promise.all([
-      ...kbnServer.newPlatform.__internals.uiPlugins.public.entries(),
-    ].map(async ([id, plugin]) => {
-      const config$ = uiPluginConfigs.get(id);
-      if (config$) {
-        return { id, plugin, config: await config$.pipe(take(1)).toPromise() };
-      } else {
-        return { id, plugin, config: {} };
-      }
-    }));
-
-    const response = h.view('ui_app', {
```

Comment:
Can we delete the views/ui_app.pug and views/chrome.pug files?

Reply:
Is this file used https://github.com/restrry/kibana/blob/290d348996c52dfb2d68e24c1f08d332f978e68b/src/legacy/server/views/index.pug ? If not or if we move it to NP, we can get rid of `pug` dependency. I believe @elastic/kibana-security would be happy

Reply:
BTW removing pug could be a breaking change for 3rd party plugins, so we should mention it in our docs.

Reply:
I'll remove the views, and create an issue to remove pug.